appcelerator/titanium_mobile
# feat: add Ti.UI.OptionBar

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28317

**Summary:**
- On iOS, the `Ti.UI.OptionBar` works just like `Ti.UI.TabbedBar` and shows a segmented control.
- On Android, this supports a "style" property which can be set to...
  * `Ti.UI.OPTION_STYLE_BUTTON` to show a material toggle button group. (See: [link](https://material.io/components/buttons/android#toggle-button))
  * `Ti.UI.OPTION_STYLE_CHIP` to show a "Chip Choice" group. (See: [link](https://material.io/components/chips#choice-chips))
  * `Ti.UI.OPTION_STYLE_RADIO` to show a radio button group. (See: [link](https://material.io/develop/android/components/radiobuttons))

**Example:**
The below is what the new `Ti.UI.OptionBar` looks like on Android and iOS with all 3 styles. Note that iOS ignores the "style" property and always shows the segmented control instead.
![OptionBar-Android](https://user-images.githubusercontent.com/26234558/104796873-c949ad00-576e-11eb-8741-1062d59cbb9e.png) <img width="295" alt="OptionBar-iOS" src="https://user-images.githubusercontent.com/26234558/104796876-ccdd3400-576e-11eb-9043-d36449d27f82.png">

**Test:**
1. Create a Classic project using `OptionBarTest.js` from [TIMOB-28317](https://jira.appcelerator.org/browse/TIMOB-28317).
2. Add the below `<application/>` element to the "tiapp.xml" file.
3. Build and run on Android.
4. Verify that the appears as shown in the screenshot above.
5. Tap on "Button 2" and verify the following gets logged.
`@@@ Button click index: 1`
6. Tap on "Chip 2" and verify the following gets logged.
`@@@ Chip click index: 1`
7. Tap on horizontal "Radio 2" and verify the following gets logged.
`@@@ Horizontal radio click index: 1`
8. Tap on vertical "Radio 2" and verify the following gets logged.
`@@@ Vertical radio click index: 1`
9. Build and run on iOS.
10. Verify the app looks like the iOS screenshot above.
11. Tap on "Button 2" and verify the following gets logged.
`@@@ Button click index: 1`

```xml
<ti:app>
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest>
			<application android:theme="@style/Theme.MaterialComponents.Light.DarkActionBar"/>
		</manifest>
	</android>
</ti:app>
```
